client.go

```diff
@@ -231,6 +234,11 @@ func NewClient(reg Registrar, conn Connector) Client {
 	}
 }
 
+// GetRegistrar returns the registrar that is registered in the client
+func (c *client) GetRegistrar() Registrar {
+	return c.registrar
```

Comment:
How unsafe is this?

Is it possible to return a copy?

Reply:
Registrar is an Interface which provides readOnly methods. So it is safe to expose.